static/js/place/chart_embed.tsx

```diff
@@ -92,22 +94,110 @@ class ChartEmbed extends React.Component<unknown, ChartEmbedStateType> {
     dataCsv: string,
     chartWidth: number,
     chartHeight: number,
+    chartHtml: string,
     chartTitle: string,
     chartDate: string,
     sources: string[]
   ): void {
     this.setState({
       modal: true,
-      svgXml: svgXml,
-      dataCsv: dataCsv,
-      chartWidth: chartWidth,
-      chartHeight: chartHeight,
-      chartTitle: chartTitle,
-      chartDate: chartDate,
-      sources: sources,
+      svgXml,
+      dataCsv,
+      chartWidth,
+      chartHeight,
+      chartHtml,
+      chartTitle,
+      chartDate,
+      sources,
     });
   }
 
+  /**
+   * Decorates chartHtml with title and provenance information embeded in an enclosing
+   * SVG node. Returns the SVG contents as a string.
+   */
+  private decorateChartHtml(): string {
+    const container = this.svgContainerElement.current;
```

Comment:
can we make a helper for all the shared functionality between decorateChartHtml and decorateSvgChart?